### ISSUE TITLE
Add symmetric member of linear operators

### ISSUE THREAD

**Issue by mrava87** [equinor/pylops#143]
- Add a boolean member called `symmetric` that defines is the operator is symmetric or not. This will be always False if the operator is not square, while it can be fixed for a square operator or depend to its input parameters like in the case of `Block` and `BlockDiagonal` operators
- Add also a 'validation' method to `LinearOperator` called `isymmetric` which validates the following equality `<x, Ay>==<y, Ax>`
- Check `symmetric` when solving an inverse problem in one of the classes that invoke script solvers and use it to choose the most appropriate solver (e.g., `cg` for symmetric operators)

**Comment by rohanbabbar04** [comment]
If this issue is still open, would like to work on it..

**Comment by mrava87** [comment]
Yes, this is open :) I would suggest to start looking into it but wait to work actively as most of the changes will be done in the `linearoperator.py` file so it is wise to wait until we approve this PR https://github.com/PyLops/pylops/pull/482

**Comment by rohanbabbar04** [comment]
Noted currently I am working on issue #386

**Comment by cako** [comment]
Maybe a good time to get back to this
